src/test_workflow/perf_test/perf_test_cluster.py

```diff
@@ -57,25 +45,22 @@ def start(self):
         subprocess.check_call(command, cwd=os.getcwd(), shell=True)
         with open(self.output_file, "r") as read_file:
             load_output = json.load(read_file)
-        self.private_ip = load_output[self.stack_name]["PrivateIp"]
-        logging.info(f"Private IP: {self.private_ip}")
-        self.public_ip = load_output[self.stack_name].get("PublicIp", None)
+            self.create_endpoint(load_output)
 
-    def endpoint(self):
-        if self.cluster_endpoint is None:
-            scheme = "https://" if self.security else "http://"
-            # If instances are configured to have public ip, use that instead.
-            host = self.private_ip if self.public_ip is None else self.public_ip
-            if host is not None:
-                self.cluster_endpoint = "".join([scheme, host, ":", str(self.port())])
-        return self.cluster_endpoint
+    @abc.abstractmethod
+    def create_endpoint(self, cdk_output):
+        pass
 
+    @property
+    def endpoint_with_port(self):
+        return self.cluster_endpoint_with_port
+
+    @property
     def port(self):
-        self.cluster_port = 443 if self.security else 9200
-        return self.cluster_port
+        return 443 if self.cluster_config.security else 80
 
     def terminate(self):
-        os.chdir(os.path.join(self.current_workspace, self.work_dir))
+        os.chdir(self.work_dir)
```

Comment:
terminate can also be converted to `@property` right?

Reply:
It shouldn't. A property is a field/noun (e.g. `object.color` or `object.shape`), invoked without `()` vs. `terminate` is an action/verb, invoked with `()` (e.g. `object.run()`, `object.stop()`).